src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,88 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
 public class SalaryInfo {
+
+    static final class DataRow {
+        private final LocalDate date;
+        private final String name;
+        private final int hours;
+        private final int rate;
+
+        DataRow(LocalDate date, String name, int hours, int rate) {
+            this.date = date;
+            this.name = name;
+            this.hours = hours;
+            this.rate = rate;
+        }
+
+        public LocalDate date() {
+            return date;
+        }
+
+        public String name() {
+            return name;
+        }
+
+        public int hours() {
+            return hours;
+        }
+
+        public int rate() {
+            return rate;
+        }
+    }
+
+    public static final DateTimeFormatter DDMMYYYY = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
+    public static DataRow parse(String str) {
```

Comment:
![image](https://user-images.githubusercontent.com/20326858/234543197-ed9eab6b-a433-46e7-af97-3305df447511.png)
Method should not be **static**
Method should be **private**
Private ones (methods, classes) should be **after** public ones.
```suggestion
    private DataRow parse(String line) {
```